view/frontend/web/js/view/payment/adyen-methods.js

```diff
@@ -68,13 +72,30 @@ define(
         /** Add view logic here if needed */
         return Component.extend({
             initialize: function () {
+                var self = this;
+
                 this._super();
 
-                adyenPaymentService.retrieveAvailablePaymentMethods().done(function (response) {
+                if (this.isGooglePayEnabled()) {
+                    var googlepayscript = document.createElement('script');
+                    googlepayscript.src = "https://pay.google.com/gp/p/js/pay.js";
+                    googlepayscript.type = "text/javascript";
+                    document.head.appendChild(googlepayscript);
+                }
 
+                if (customer.isLoggedIn()) {
+                    self.setAdyenPaymentMethods();
+                }
+
+                quote.shippingAddress.subscribe(function() {
+                    self.setAdyenPaymentMethods();
```

Comment:
check if the country is changed and don't reset it every time when the shipping address is changed